mmv1/products/firestore/BackupSchedule.yaml

```diff
@@ -0,0 +1,114 @@
+  The actual backups are not though.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/firestore/docs/backups'
+  api: 'https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases.backupSchedules'
+skip_sweeper: true
+id_format: 'projects/{{project}}/databases/{{database}}/backupSchedules/{{name}}'
+import_format:
+  - 'projects/{{project}}/databases/{{database}}/backupSchedules/{{name}}'
+  - '{{project}}/{{database}}/{{name}}'
+  - '{{database}}/{{name}}'
+docs: !ruby/object:Provider::Terraform::Docs
+  warning: |
+    This resource creates a Firestore Backup Schedule on a project that already has
+    a Firestore database. If you haven't already created it, you may
+    create a `google_firestore_database` resource with `type` set to
+    `"FIRESTORE_NATIVE"` and `location_id` set to your chosen location.
+    If you wish to use App Engine, you may instead create a
+    `google_app_engine_application` resource with `database_type` set to
+    `"CLOUD_FIRESTORE"`. Your Firestore location will be the same as
+    the App Engine location specified.
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'firestore_backup_schedule_daily'
+    primary_resource_id:
+      'daily-backup'
+    test_env_vars:
+      project_id: :FIRESTORE_PROJECT_NAME
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'firestore_backup_schedule_weekly'
+    primary_resource_id:
+      'weekly-backup'
+    test_env_vars:
+      project_id: :FIRESTORE_PROJECT_NAME
+parameters:
+  - !ruby/object:Api::Type::String
+    name: database
+    description: |
+      The Firestore database id. Defaults to `"(default)"`.
+    default_value: '(default)'
+    immutable: true
+    url_param_only: true
+properties:
+  - !ruby/object:Api::Type::String
+    name: name
+    output: true
+    description: |
+      A server defined name for this index. Format:
+      `projects/{{project}}/databases/{{database}}/backupSchedules/{{backupSchedule}}
+    immutable: true
+    custom_flatten: templates/terraform/custom_flatten/name_from_self_link.erb
+  - !ruby/object:Api::Type::String
+    name: retention
+    required: true
+    description: |
+      At what relative time in the future, compared to its creation time, the backup should be deleted, e.g. keep backups for 7 days.
+      A duration in seconds with up to nine fractional digits, ending with 's'. Example: "3.5s".
+
+      For a daily backup recurrence, set this to a value up to 7 days. If you set a weekly backup recurrence, set this to a value up to 14 weeks.
+  - !ruby/object:Api::Type::NestedObject
+    name: dailyRecurrence
+    description: |
+      For a schedule that runs daily at a specified time.
+    exactly_one_of:
+      - daily_recurrence
+      - weekly_recurrence
+    send_empty_value: true
+    allow_empty_object: true
+    immutable: true
```

Comment:
When trying to patch an existing backup schedule from daily to weekly, it has no effect.

I've created a backup schedule with daily recurrence.
And as you can see in the example below, I've set the schedule to weekly but the schedule is still daily in the response:

<img width="404" alt="image" src="https://github.com/GoogleCloudPlatform/magic-modules/assets/10195482/1427935e-902f-4816-af91-89871bd37947">

---

Regarding the question of what happens with the backups when the backup schedule is deleted, let me say the following:

The docs says that:
> Deleting the source database does not automatically delete related backups.
[source](https://cloud.google.com/firestore/docs/backups#:~:text=Backups%20have%20a%20configurable%20retention%20period%20and%20are%20stored%20until%20the%20retention%20period%20expires%20or%20until%20you%20delete%20the%20backup.%20Deleting%20the%20source%20database%20does%20not%20automatically%20delete%20related%20backups.)

This means backups will persist until reaching the end of the retention period or when [manually deleting the backup](https://cloud.google.com/firestore/docs/reference/rest/v1/projects.locations.backups/delete).

While the docs do not mention what happens when only the backup schedule is deleted, I strongly think that the same applies to backup schedules.
Deleting a schedule does NOT delete the backups created using this backup schedule.

Reply:
I can verify the exact behavior tomorrow though when I finally have a backup and can delete the schedule again.

Reply:
Brilliant, thank you!

If the update truly doesn't work, despite there being no API errors, then forcing recreation of the resource in Terraform makes sense. But we definitely want to avoid any unintended consequences of that.

If the update does work but has an 'eventual consistency' issue then there might be different approaches to the problem. If you wait a few minutes and then GET the backup schedule you updated, do the new recurrence settings show up?

Reply:
### Regarding changing the recurrence:

Even one day later, the recurrence is still daily, so there is no eventual consistency thing going on here.
It seems like updating the recurrence without deleting and re-creating is not supported.

### Regarding the backups:

These are the requests I've done:

`GET /projects/<project>/locations/-/backups`
![image](https://github.com/GoogleCloudPlatform/magic-modules/assets/10195482/b2131049-2557-472f-abaa-86e93fd5f5b6)

`GET /projects/<project>/databases/<database>/backupSchedules`
![image](https://github.com/GoogleCloudPlatform/magic-modules/assets/10195482/d16c7a5e-8df2-4899-86dc-42fb6001e91f)

`DELETE /projects/<project>/databases/<database>/backupSchedules/<schedule>`
![image](https://github.com/GoogleCloudPlatform/magic-modules/assets/10195482/2ecf6afc-aa1c-40b4-93e0-e7749dc8c4fa)

`GET /projects/<project>/databases/<database>/backupSchedules`
![image](https://github.com/GoogleCloudPlatform/magic-modules/assets/10195482/b530ce34-db4a-47a2-a4f1-f75845953fbb)

`GET /projects/<project>/locations/-/backups`
![image](https://github.com/GoogleCloudPlatform/magic-modules/assets/10195482/591cd7c3-7727-42da-93f4-c701259f0860)

So even after deleting the backup schedule the backup is still there!
Meaning deletion of the schedule should have no issue or should not need any special consideration.

We may want to expose the backups as data source in the future potentially though.

Reply:
@SarahFrench does that answer your questions?

Reply:
(Firestore developer speaking) Changing an existing backup schedule from weekly to daily (or vice-versa) is not supported. If you want to make such a change, you can delete the weekly schedule and then create a new one with daily recurrence.

Reply:
Perfect then the current behavior should be fine.

Essentially if the recurrence changes the resource has to be deleted and re-created.

-- 

@rwhogg this behavior should not be a problem as previous backups done with a different schedule/recurrence are kept even though the backup schedule has been deleted, no?

Reply:
I'm going to mark this thread as resolved, as we now know that the recurrence definitely can't be updated and from your [manual experiments](https://github.com/GoogleCloudPlatform/magic-modules/pull/8908#discussion_r1345910631) there doesn't seem to be a risk of accidentally causing users to lost backups.